### PR TITLE
opt: add session setting to disable inequality lookup joins

### PR DESCRIPTION
**Pull request by DrewKimball**
This commit adds a session setting (default on) that can be toggled to off in order to disable the lookup join behavior added in #85597. This will provide a quick route to mitigate any customer issues that might be caused by #85597.

Release note: None

### CODE REVIEW

**Review by mgartner — APPROVED** [review]
:lgtm: Thanks for doing this!

> Reviewed 10 of 10 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88884#-:-ND3WjbZA64U2ybQ8UdC:b-pt8m11)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @DrewKimball)

___
*[`pkg/sql/vars.go` line 2299 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/88884#-ND3V6p-5RM_PifkNs_P:-ND3V6p03D10BIc-CnKS:bchyejl) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4f15ef6ab157f0a6d2521ec0bfa93fa4e5eafe8/pkg/sql/vars.go#L2299)):*
> ```Go
> 
> 	// CockroachDB extension.
> 	`inequality_lookup_join_enabled`: {
> ```

nit: should we name this `variable_inequality_lookup_join_enabled` (and variables/fields related to this setting) to make distinguish it from inequalities with one variable and a constant? Maybe that's too wordy though...
___
*[`pkg/sql/logictest/testdata/logic_test/lookup_join` line 1483 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/88884#-ND3WMVl5UMFCDTwPdXh:-ND3WMVl5UMFCDTwPdXi:bgckk41) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4f15ef6ab157f0a6d2521ec0bfa93fa4e5eafe8/pkg/sql/logictest/testdata/logic_test/lookup_join#L1483)):*
> ```
> 
> query T
> EXPLAIN SELECT * FROM abc INNER LOOKUP JOIN def_e_decimal ON f = b AND e <= a::DECIMAL ORDER BY a, b, c, d, e, f
> ```

nit: we usually put `EXPLAIN` tests in `execbuilder` tests so that any metamorphic settings can't make the test flakey

**Review by DrewKimball — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88884)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @mgartner)

___
*[`pkg/sql/vars.go` line 2299 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/88884#-ND3V6p-5RM_PifkNs_P:-NDAlGgF6ZB3FkkQMYmF:b-odhds1) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4f15ef6ab157f0a6d2521ec0bfa93fa4e5eafe8/pkg/sql/vars.go#L2299)):*
<details><summary><i>Previously, mgartner (Marcus Gartner) wrote…</i></summary><blockquote>

nit: should we name this `variable_inequality_lookup_join_enabled` (and variables/fields related to this setting) to make distinguish it from inequalities with one variable and a constant? Maybe that's too wordy though...
</blockquote></details>

I think it's fine, Done.
___
*[`pkg/sql/logictest/testdata/logic_test/lookup_join` line 1483 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/88884#-ND3WMVl5UMFCDTwPdXh:-NDAnwFk68pgFEZtLmx0:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/b4f15ef6ab157f0a6d2521ec0bfa93fa4e5eafe8/pkg/sql/logictest/testdata/logic_test/lookup_join#L1483)):*
<details><summary><i>Previously, mgartner (Marcus Gartner) wrote…</i></summary><blockquote>

nit: we usually put `EXPLAIN` tests in `execbuilder` tests so that any metamorphic settings can't make the test flakey
</blockquote></details>

Done.

**Comment by DrewKimball** [comment]
TFTR!

**Review by yuzefovich — APPROVED** [review]
:lgtm:

> Reviewed 1 of 10 files at r1, 14 of 14 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/88884#-:-NDBK-Bb4yoKkM8KAJMS:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (and 1 stale) (waiting on @mgartner)

**Comment by DrewKimball** [comment]
Test failure looks like a flake.

bors r+

**Comment by DrewKimball** [comment]
blathers backport 22.2.0
